legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/data/core/EmbeddedDataCompilerHelper.java

```diff
@@ -90,4 +93,11 @@ else if (embeddedData instanceof DataElementReference)
             return null;
         }
     }
+
+    private static boolean validatePairForModelStoreData(Pair value, CompileContext context)
+    {
+       return (value.first instanceof PackageableElementPtr && value.second instanceof PackageableElementPtr &&
+                context.pureModel.getPackageableElement(((PackageableElementPtr) value.first).fullPath) instanceof Root_meta_external_shared_format_binding_Binding &&
+                context.pureModel.getPackageableElement(((PackageableElementPtr) value.second).fullPath) instanceof Root_meta_pure_data_DataElement);
```

Comment:
do we also need to check that data is of external format type and not let say relational?

Reply:
it does fail when we try executing, but I think it would be a good idea, to throw the error on compiler stage

Reply:
Resolved